src/views/List.jsx

```diff
@@ -122,6 +140,42 @@ export function List({ data, listToken, setListToken, user }) {
 				});
 		}
 	};
+	const listOfShoppingListItems = [];
+
+	// this is not inside a function?
+	!exclude
+		? renderedList.forEach((item) => {
+				listOfShoppingListItems.push(item.name);
+		  })
+		: excludeChecked(renderedList).forEach((item) => {
+				listOfShoppingListItems.push(item.name);
+		  });
+
+	const handleCalendarDownload = (evt) => {
+		const defaultStartTime = new Date(Date());
+		const dt = new Date();
+		dt.setHours(dt.getHours() + 1);
+		const defaultEndTime = dt;
+		const eventConfig = {
+			title: 'Shopping trip',
+			description: listOfShoppingListItems.join('\n'),
+			start: defaultStartTime,
+			end: defaultEndTime,
+		};
+		if (evt.target.name === 'ical') {
+			const ical = new ICalendar(eventConfig);
+			ical.download();
+		} else if (evt.target.name === 'google') {
+			const googleCalendar = new GoogleCalendar(eventConfig);
+			window.open(googleCalendar.render());
+		} else if (evt.target.name === 'outlook') {
+			const outlookCalendar = new OutlookCalendar(eventConfig);
+			window.open(outlookCalendar.render());
+		} else {
+			const yahooCalendar = new YahooCalendar(eventConfig);
+			window.open(yahooCalendar.render());
+		}
```

Comment:
`switch` case here maybe easier?

Note, maybe this function and the other calendar functions can be placed in a `calendar.tsx` helper!